ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.cpp

```diff
@@ -370,6 +370,17 @@ ttnn::Tensor ReshapeViewOperation::invoke(
 
     const uint32_t shape_second_last_dim = shape.rank() >= 2 ? shape[-2]:1;
     const uint32_t tensor_shape_second_last_dim = tensor_shape.rank() >= 2 ? tensor_shape[-2]:1;
+
+    // Just edit shape if shape has a 0 dimension
+    if (tensor.volume() == 0) {
```

Comment:
Actually `tensor.volume()` refers to physical volume, not logical volume. You can use either `tensor.get_logical_volume()` or `tensor.logical_shape().volume()`
cc @ayerofieiev-tt